### PR TITLE
Wishlist and Cart

### PR DESCRIPTION
**Pull request by AbhishekkGautam**
### Wishlist
- product card with image, title, price, discount, rating & brand name.
- remove product using the close icon.
- move product to cart using the `move to cart` button.
- you can see the wishlist product count in the navbar icon badge. 
- if the same item is already present in the cart, then clicking on the `move to cart` button will increase the quantity.
- user can view the wishlist only after logging in. 

### Cart
- product card with image, title, price, discount, & quantity counter.
- `remove` button 
- `save for later` button
- cart summary card 
  - total items count
  - total MRP
  - total discount
  - delivery charges
  - the total amount to pay. 
  - `proceed to pay` button

### Dummy Login Credentials
Email: `john@gmail.com`
Password: `johndoe`

### CODE REVIEW

**Review by vijaytembugade — APPROVED** [review]
Great work . Clean and neat code.

File: src/context/AuthContext.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ const AuthProvider = ({ children }) => {
           type: "SAVE_USER_INFO",
           payload: foundUser,
         });
-        navigate("/");
+        navigate(-1);
```
Comment (pritam-kr) [inline src/context/AuthContext.js:65]: why navigate is -1 ??

File: src/context/AuthContext.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ const AuthProvider = ({ children }) => {
...
       }
     } catch (error) {
       dispatch({ type: "AUTH_ERROR", payload: error.response });
```
Comment (pritam-kr) [inline src/context/AuthContext.js:68]: wow, even you used reducer for error.

Comment (pritam-kr) [reply src/context/AuthContext.js:68]: I learned while reviewing this.

File: src/helpers/helpers.js [hunk 3]
```diff
@@ -91,3 +91,36 @@ export const getFilteredProducts = (products, state) => {
...
   return tempProducts;
 };
+
+export const totalItemsInCart = cart => {
+  return [...cart].reduce((acc, curr) => (acc = acc + curr.qty), 0);
+};
+
+export const getTotalMrpInCart = cart => {
+  return [...cart].reduce(
+    (acc, curr) => (acc = acc + curr.priceInMrp * curr.qty),
```
Comment (pritam-kr) [inline src/helpers/helpers.js:101]: why you're spreading array ???

File: src/pages/Cart/Cart.css [hunk 4]
```diff
@@ -0,0 +1,233 @@
+@import "../../index.css";
+
+.cart-section {
+  margin-top: 6rem;
+}
+
+.cart-container {
+  padding: 0 1rem;
+}
+
```
Comment (pritam-kr) [inline src/pages/Cart/Cart.css:10]: There is no gap..👍😁

**Review by pritam-kr — APPROVED** [review]
Everything looks clean, the UI part is also great. Also, I have added some points. please explain.

**Comment by pritam-kr** [comment]
Hey, Quantity decrement for the product going lower than 1.

File: src/context/AuthContext.js (hunk 1 above)
Comment (AbhishekkGautam) [reply src/context/AuthContext.js:65]: If user visits the cart route & is not logged in. So user will click on login button & navigate to login page. After logging in, user will navigate back to cart page. 
User will navigate back to the same page where he/she was before logging in. ( UX stuff )

File: src/helpers/helpers.js (hunk 3 above)
Comment (AbhishekkGautam) [reply src/helpers/helpers.js:101]: Just to avoid doing any thing on the original array.

**Comment by AbhishekkGautam** [comment]
> Hey, Quantity decrement for the product going lower than 1.

Will look into it. Thank you

**Pull request merged**